source_code/main_mcu/src/COMMS/comms_aux_mcu.c

```diff
@@ -243,7 +244,112 @@ void comms_aux_mcu_deal_with_received_event(aux_mcu_message_t* received_message)
+
+    /* Set same message type and payload length. We are using the same message
+     * in both directions
+     */
+    send_message->message_type = AUX_MCU_MSG_TYPE_FIDO2;
+    send_message->FIDO2_message.message_type = AUX_MCU_MSG_TYPE_AUTH_CRED_RSP;
+    send_message->payload_length1 = sizeof(FIDO2_message_t);
+    return FIDO2_MSG_RCVD;
+}
+
+/*! \fn     comms_aux_mcu_handle_FIDO2_make_auth_data_msg(FIDO2_message_t const *received_message, aux_mcu_message_t *send_message)
+*   \brief  routine handling authenticating a credential
+*   \param  received_message The received message
+*   \param  send_message The response message
+*   \return FIDO2_MSG_RCVD
+*/
+static comms_msg_rcvd_te comms_aux_mcu_handle_FIDO2_make_auth_data_msg(FIDO2_message_t const *received_message, aux_mcu_message_t *send_message)
+{
+    FIDO2_make_auth_data_req_message_t const *request = &received_message->FIDO2_make_auth_data_req_message;
+    FIDO2_make_auth_data_rsp_message_t *response = &send_message->FIDO2_message.FIDO2_make_auth_data_rsp_message;
+
+    fido2_process_make_auth_data(request, response);
+
+    send_message->message_type = AUX_MCU_MSG_TYPE_FIDO2;
+    send_message->FIDO2_message.message_type = AUX_MCU_MSG_TYPE_MAD_RSP;
+    send_message->payload_length1 = sizeof(FIDO2_message_t);
+    return FIDO2_MSG_RCVD;
+}
+
+/*! \fn     comms_aux_mcu_handle_FIDO2_GNC_(FIDO2_message_t const *received_message, aux_mcu_message_t *send_message)
+*   \brief  routine handling get next credential message
+*   \param  received_message The received message
+*   \param  send_message The response message
+*   \return FIDO2_MSG_RCVD
+*/
+static comms_msg_rcvd_te comms_aux_mcu_handle_FIDO2_GNC_msg(FIDO2_message_t const *received_message, aux_mcu_message_t *send_message)
+{
+    FIDO2_get_next_credential_req_message_t const *request = &received_message->FIDO2_get_next_credential_req_message;
+    FIDO2_get_next_credential_rsp_message_t *response = &send_message->FIDO2_message.FIDO2_get_next_credential_rsp_message;
+
+    fido2_process_get_next_credential(request, response);
+
+    send_message->message_type = AUX_MCU_MSG_TYPE_FIDO2;
+    send_message->FIDO2_message.message_type = AUX_MCU_MSG_TYPE_GNC_RSP;
+    send_message->payload_length1 = sizeof(FIDO2_message_t);
+    return FIDO2_MSG_RCVD;
+}
+
+/*! \fn     comms_aux_mcu_handle_FIDO2_unknown_msg(FIDO2_message_t const *received_message, aux_mcu_message_t *send_message)
+*   \brief  routine handling unknown FIDO2 messages
+*   \param  received_message The received message
+*   \param  send_message The response message
+*   \return UNKNOW_MSG_RCVD
+*/
+static comms_msg_rcvd_te comms_aux_mcu_handle_FIDO2_unknown_msg(FIDO2_message_t const *received_message, aux_mcu_message_t *send_message)
+{
+    return UNKNOW_MSG_RCVD;
+}
+
+typedef comms_msg_rcvd_te (*handle_FIDO2_msg_t)(FIDO2_message_t const *received_message, aux_mcu_message_t *send_message);
+
+static handle_FIDO2_msg_t FIDO2_message_handlers[] = {
```

Comment:
from a security point of view i'm really not a big fan of an array of functions without the correct static_assert that makes sure that AUX_MCU_MSG_TYPE_FIDO2_END-AUX_MCU_MSG_TYPE_FIDO2_START is equal to the number of items in FIDO2_message_handlers.
I can fully understand a switch case isn't the prettiest thing.... it is however the thing with less room for error... though granted we're only talking about 6 items there...

Reply:
I'll convert to switch! No problem.